src/borg/testsuite/repository.py

```diff
@@ -15,11 +14,24 @@
 from ..remote import RemoteRepository, InvalidRPCMethod, PathNotAllowed
 from ..repository import Repository, LoggedIO, MAGIC, MAX_DATA_SIZE, TAG_DELETE, TAG_PUT2, TAG_PUT, TAG_COMMIT
 from ..repoobj import RepoObj
-from . import BaseTestCase
 from .hashindex import H
 
 
-UNSPECIFIED = object()  # for default values where we can't use None
+@pytest.fixture()
+def repository(tmpdir):
+    repository_location = os.path.join(str(tmpdir), "repository")
+    repository = Repository(repository_location, exclusive=True, create=True)
+    yield repository
+
+
+@contextmanager
+def reopen(repository, exclusive=True):
+    repo = Repository(repository.path, exclusive=exclusive, create=False)
+    repo.__enter__()
+    try:
+        yield repo
+    finally:
+        repo.__exit__(None, None, None)
```

Comment:
hmm, guess you have 1 context manager too much here:
- the Repository instance you have in `repo` is already a CM (== it has `__enter__` and `__exit__` methods)
- you are creating another CM here, wrapping the other one and doing nothing else

guess that would also work:
```
def reopen(repository, exclusive=True):
    return Repository(repository.path, exclusive=exclusive, create=False)


def test(repository):
    with repository:
        repository.do_stuff()
    with reopen(repository) as repository:
        repository.check_stuff()
```

Reply:
@ThomasWaldmann Thank you for your patience as I work to understand CM's better. I see now why the repo's `__enter__` and `__exit__` weren't getting called automatically during the `reopen` the same way they were with the original repository fixture, because I had wrapped them in an additional CM. Thank you for clarifying that.